apis/v1alpha1/gateway_types.go

```diff
@@ -347,6 +341,7 @@ type GatewayTLSConfig struct {
 // TLSModeType type defines behavior of gateway with TLS protocol.
 // +kubebuilder:validation:Enum=Terminate;Passthrough
 // +kubebuilder:default=Terminate
+// +kubebuilder:validation:MaxLength=30
 type TLSModeType string
```

Comment:
nit: 30 characters is excessive (unless we were naming in Java) for a ModeType. How about 16? Is that too less?

Reply:
Good call, removed in favor of enum validation.